### ISSUE TITLE
Dots break parameter substitution in FT.SEARCH

### ISSUE THREAD

**Issue by hackerfriendly** [RediSearch/RediSearch#3549]
I'm running Redis 7.0.10 / RediSearch v2.6.9.

Searching for strings containing a . works as expected:

```
> hset paramsDemo:noDot host "example.com"
(integer) 1

> FT.CREATE paramsDemo on HASH PREFIX 1 paramsDemo: SCHEMA host TEXT
"OK"

> FT.SEARCH paramsDemo @host:example.com
1) "1"
2) "paramsDemo:noDot"
3) 1) "host"
   2) "example.com"
```

...but not when using PARAMS:

```
> FT.SEARCH paramsDemo @host:$host DIALECT 2 PARAMS 2 host example.com
1) "0"
```

Search still works with PARAMS as long as it doesn't contain a dot:

```
> FT.SEARCH paramsDemo @host:$host DIALECT 2 PARAMS 2 host example
1) "1"
2) "paramsDemo:noDot"
3) 1) "host"
   2) "example.com"
```

No variation of escaping or quoting the parameter seems to remedy this.

**Comment by hackerfriendly** [comment]
I see, this seems to be a tokenization issue.

**Comment by oshadmi** [comment]
@hackerfriendly Thank you for reporting. Why did you close this issue?

**Comment by oshadmi** [comment]
Could be related to https://github.com/RediSearch/RediSearch/pull/3403/files#r1115307827

**Comment by hackerfriendly** [comment]
I'm new to Redis, and when I filed the issue I didn't realize that TEXT fields always tokenize. I changed the field to TAG to force a literal match, which works for my application. So I closed the issue.

But after looking at FT.EXPLAIN it's clear that the same string is being treated differently if it's literal or a parameter.

Is it possible to control the tokenization of TEXT fields? I couldn't find anything relevant in the docs. I would expect the . delimiter to only tokenize if there is whitespace on either side, otherwise it's not clear how one could match common things like host names or IP addresses.
